sql: TestCancelQueryPermissions flaking

The `TestCancelQueryPermissions` flaked with:

```
=== RUN   TestCancelQueryPermissions/unpermissioned_users_cannot_cancel_other_users
panic: Fail in goroutine after TestCancelQueryPermissions/non-admins_with_CANCELQUERY_can_cancel_non-admins has completed

goroutine 365437 [running]:
testing.(*common).Fail(0xc00757e780)
	/usr/local/go/src/testing/testing.go:688 +0x125
testing.(*common).FailNow(0xc00757e780)
	/usr/local/go/src/testing/testing.go:710 +0x2b
testing.(*common).Fatalf(0xc00757e780, 0x47c9e4c, 0x1c, 0xc00ba7a4a0, 0x2, 0x2)
	/usr/local/go/src/testing/testing.go:806 +0x93
github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).ExpectErr(0xc002060fa8, 0x560fca0, 0xc00757e780, 0x47b8f45, 0x18, 0x47b6bed, 0x18, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/sqlutils/sql_runner.go:97 +0x1c5
github.com/cockroachdb/cockroach/pkg/sql_test.TestCancelQueryPermissions.func2.1(0xc0136594f0, 0xc002586340, 0xc00757e780)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/run_control_test.go:327 +0xae
created by github.com/cockroachdb/cockroach/pkg/sql_test.TestCancelQueryPermissions.func2
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/run_control_test.go:318 +0x13d
```

Flake can be found at - https://teamcity.cockroachdb.com/viewLog.html?buildId=2876992&buildTypeId=Cockroach_UnitTests

Not skipping for now, as we have only seen one instance of it.

--- a/pkg/sql/run_control_test.go
+++ b/pkg/sql/run_control_test.go
@@ -17,6 +17,7 @@ import (
 	"math/rand"
 	"net/url"
 	"strings"
+	"sync"
 	"testing"
 	"time"
 
@@ -25,7 +26,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sqltestutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -242,7 +242,6 @@ GRANT admin TO has_admin2;
 
 func TestCancelQueryPermissions(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	skip.WithIssue(t, 63547, "https://cockroachlabs.slack.com/archives/C016CAD2HQ8/p1627290710022700")
 	defer log.Scope(t).Close(t)
 
 	// getQueryIDs retrieves the IDs of any currently running queries for the
@@ -312,8 +311,13 @@ GRANT admin TO has_admin2;
 		{"unpermissioned users cannot cancel other users", "no_perms", "has_cancelquery", false,
 			"this operation requires CANCELQUERY privilege"},
 	}
+	// Avoid using subtests with t.Run since we may need to access `t` after the
+	// subtest is done. Use a WaitGroup to make sure the error from the pg_sleep
+	// goroutine is checked.
+	wg := sync.WaitGroup{}
 	for _, tc := range testCases {
-		t.Run(tc.name, func(t *testing.T) {
+		func() {
+			wg.Add(1)
 			// Start a query with the target user.
 			targetDB := getUserConn(t, tc.targetUser, testCluster.Server(0))
 			defer targetDB.Close()
@@ -326,7 +330,11 @@ GRANT admin TO has_admin2;
 					// end of the test.
 					errRE = "sql: database is closed"
 				}
-				sqlutils.MakeSQLRunner(targetDB).ExpectErr(t, errRE, "SELECT pg_sleep(1000000)")
+				_, err := targetDB.ExecContext(context.Background(), "SELECT pg_sleep(100)")
+				if !testutils.IsError(err, errRE) {
+					t.Errorf("expected error '%s', got: %v", errRE, err)
+				}
+				wg.Done()
 			}()
 
 			// Retrieve the query ID.
@@ -353,8 +361,10 @@ GRANT admin TO has_admin2;
 			} else {
 				runner.ExpectErr(t, tc.expectedErrRE, `CANCEL QUERY $1`, queryID)
 			}
-		})
+		}()
 	}
+	testCluster.Stopper().Stop(ctx)
+	wg.Wait()
 }
 
 func TestCancelIfExists(t *testing.T) {

